src/popup/components/Modal.vue

```diff
@@ -100,19 +104,27 @@ export default defineComponent({
   },
   emits: ['close', 'opened'],
   setup(props, { slots }) {
+    const isModalVisible = ref(props.isVisible);
```

Comment:
Turns out, since modals are only mounted when they are visible we do not need to pass the variable from the parent component to the modal (no more copying prop state).
We do need the v-if statement though to make sure that the modal plays the exit animation when it is closed.
[[see](https://v3-migration.vuejs.org/breaking-changes/transition-as-root.html)]